flashinfer/mla.py

```
@@ -767,9 +769,13 @@ def xqa_batch_decode_with_kv_cache_mla(
             f"XQA MLA only supports q_len_per_request == 1, got {q_len_per_request}"
         )
-    if query.dtype != torch.float8_e4m3fn or kv_cache.dtype != torch.float8_e4m3fn:
+    fp8_ok = (
+        query.dtype == torch.float8_e4m3fn and kv_cache.dtype == torch.float8_e4m3fn
+    )
+    bf16_ok = query.dtype == torch.bfloat16 and kv_cache.dtype == torch.bfloat16
+    if not (fp8_ok or bf16_ok):
         raise ValueError(
-            f"XQA MLA only supports fp8 tensor core operation, got {query.dtype} and {kv_cache.dtype}"
+            f"XQA MLA supports (fp8, fp8) or (bfloat16, bfloat16) only, got {query.dtype} and {kv_cache.dtype}"
         )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)